ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -2885,3 +2885,59 @@ def test_ipahealthcheck_iparaagent_bad_serial(self, update_ra_cert_desc):
                 assert check["kw"]["msg"] == error_msg
             else:
                 assert error_reason in check["kw"]["msg"]
+
+
+class TestIpaHealthCheckSingleMaster(IntegrationTest):
+
+    @classmethod
+    def install(cls, mh):
+        # Nota Bene: The ipa server is not installed
+        tasks.install_packages(cls.master, HEALTHCHECK_PKG)
+
+    def test_ipahealthcheck_mismatching_certificates_subject(self):
+        """
+        Test if healthcheck uses cert subject base from IPA and not from
+        REALM. This prevents false-positive errors when the subject base is
+        customized.
+
+        Related: https://github.com/freeipa/freeipa-healthcheck/issues/253
+        """
+        # install master with custom cert subject base
+        tasks.install_master(
+            self.master,
+            setup_dns=True,
+            extra_args=[
+                '--no-dnssec-validation',
+                '--subject-base=O=LINUX.IS.GREAT,C=EU'
+            ]
+        )
+
+        returncode, data = run_healthcheck(self.master,
+                                           source="ipahealthcheck.ipa.certs",
+                                           check="IPADogtagCertsMatchCheck",
+                                           failures_only=True)
+        assert returncode == 0
+        assert len(data) == 0
+
+        # uninstall server for the next step
+        tasks.uninstall_master(self.master)
+
+        # install master with custom CA certificate subject DN
+        tasks.install_master(
+            self.master,
+            setup_dns=True,
+            extra_args=[
+                '--no-dnssec-validation',
+                '--ca-subject=CN=Healthcheck test,O=LINUX.IS.GREAT'
+            ]
+        )
+
+        returncode, data = run_healthcheck(self.master,
+                                           source="ipahealthcheck.ipa.certs",
+                                           check="IPADogtagCertsMatchCheck",
+                                           failures_only=True)
+        assert returncode == 0
+        assert len(data) == 0
+
+        # cleanup
+        tasks.uninstall_master(self.master)
```

Comment:
Looks good, just one nit. I think this should either be wrapped in a try/finally so that tasks.uninstall_master(self.master) is always run or use a fixture to do the uninstall (I prefer the first but line length could become an issue)

If one of the tests fails it will leave the server configured as-is which could break subsequent tests as well.